src/lib/provable/crypto/foreign-ecdsa.ts

```diff
@@ -127,6 +169,27 @@ class EcdsaSignature {
     );
   }
 
+  /**
+   * Verify the ECDSA signature given the message hash (a {@link Scalar}) and public key (a {@link Curve} point).
+   *
+   * This is a building block of {@link EcdsaSignature.verify}, where the input message is also hashed.
+   * In contrast, this method just takes the message hash (a curve scalar) as input, giving you flexibility in
+   * choosing the hashing algorithm.
+   */
+  verifySignedHashV2(
```

Comment:
Was it necessary to add a new `verifySignedHashV2` function here? It seems so since the deprecated `verify` was being used here, but not sure if it's necessary to reflect everywhere.

Reply:
yeah thanks for adding this as well!